File: examples/extract-content-from-a-url/rest/csharp/ExtractContent.cs
// Fetches the fully rendered HTML of a page using the Browserless /content endpoint.
// Uses System.Net.Http and System.Text.Json from the .NET standard library.
//
// Run: dotnet run

using System.Net.Http;
using System.Text;
using System.Text.Json;

const string Token = "YOUR_API_TOKEN_HERE";
string endpoint = $"https://production-sfo.browserless.io/content?token={Token}";

var payload = new { url = "https://example.com" };

using var client = new HttpClient();
var request = new HttpRequestMessage(HttpMethod.Post, endpoint);
request.Headers.Add("Cache-Control", "no-cache");
request.Content = new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json");

var response = await client.SendAsync(request);
string html = await response.Content.ReadAsStringAsync();
Console.WriteLine(html[..Math.Min(500, html.Length)]);

File: examples/extract-content-from-a-url/rest/java/ExtractContent.java
// Fetches the fully rendered HTML of a page using the Browserless /content endpoint.
// java.net.http is included in JDK 11+ — no extra dependency needed.
//
// Run: javac ExtractContent.java && java ExtractContent

import java.net.URI;
import java.net.http.*;

public class ExtractContent {
    private static final String TOKEN = "YOUR_API_TOKEN_HERE";

    public static void main(String[] args) throws Exception {
        String endpoint = "https://production-sfo.browserless.io/content?token=" + TOKEN;
        String jsonData = "{\"url\": \"https://scraping-sandbox.netlify.app/javascript-enabled\"}";

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Cache-Control", "no-cache")
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(jsonData))
            .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        System.out.println(response.body().substring(0, Math.min(500, response.body().length())));
    }
}
